### ISSUE TITLE
Missing `await` keyword in `authenticate()` method

### ISSUE THREAD

**Issue by ghost** [ajmas/vue-authenticate#8]
Hi there @ajmas 

While attempting to use this project with Linkedin as provider, I noticed that the `authenticate()` method always throws the `Authentication failed` error immediately. After some digging, it seems that the call to the asynchronous `init()` method of the `OAuth1` or `OAuth2` instance is not made using the `await` keyword, which leads to the presence of a token being checked before the token is received.

I think that the line:

```
const response = providerInstance.init(userData);
```

should be changed to:

```
const response = await providerInstance.init(userData);
```

in `authenticate.js`, line 243.

**Comment by ajmas** [comment]
Yup that look like a bug, especially since `init` is `async`.

**Comment by ajmas** [comment]
Addressed by PR https://github.com/ajmas/vue-authenticate-2/pull/10

**Comment by ajmas** [comment]
Available as part of 2.0.0-rc4.0
